diff --git a/cipher.c b/cipher.c
index 7d6e7d8c6..f6b40798a 100644
--- a/cipher.c
+++ b/cipher.c
@@ -121,8 +121,6 @@ cipher_alg_list(char sep, int auth_only)
 	const struct sshcipher *c;
 
 	for (c = ciphers; c->name != NULL; c++) {
-		if ((c->flags & CFLAG_INTERNAL) != 0)
-			continue;
 		if (auth_only && c->auth_len == 0)
 			continue;
 		if (ret != NULL)
@@ -224,7 +222,7 @@ ciphers_valid(const char *names)
 	for ((p = strsep(&cp, CIPHER_SEP)); p && *p != '\0';
 	    (p = strsep(&cp, CIPHER_SEP))) {
 		c = cipher_by_name(p);
-		if (c == NULL || (c->flags & CFLAG_INTERNAL) != 0) {
+		if (c == NULL) {
 			free(cipher_list);
 			return 0;
 		}
